src/journal/journald-native.c

```diff
@@ -193,14 +195,8 @@ void server_process_native_message(
 
                                 } else if (l >= 8 &&
                                            startswith(p, "MESSAGE=")) {
-                                        char *t;
-
-                                        t = strndup(p + 8, l - 8);
-                                        if (t) {
-                                                free(message);
-                                                message = t;
```

Comment:
We can't remove `strndup` because `server_forward_kmsg` and friends expect a pointer to the null-terminated string.
Try
```sh
-bash-4.3# grep Syslog /etc/systemd/journald.conf
ForwardToSyslog=yes
MaxLevelSyslog=debug

-bash-4.3# printf 'MESSAGE=TEST\n' | logger --journal

-bash-4.3# dmesg
...

[ 6826.796792] logger[12716]: TEST
               CODE_FILE=misc-utils/logger.c
               CODE_LINE=343
               CODE_FUNC=journald_entry
               SYSLOG_IDENTIFIER=logger

```
The correct output is
```
[  906.731028] logger[230]: TEST
```